src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -621,92 +559,58 @@ MulticopterPositionControl::parameters_update(bool force)
 	}
 
 	if (updated || force) {
-		/* update C++ param system */
-		updateParams();
-
-		/* update legacy C interface params */
-		param_get(_params_handles.thr_min, &_params.thr_min);
-		param_get(_params_handles.thr_max, &_params.thr_max);
-		param_get(_params_handles.thr_hover, &_params.thr_hover);
-		_params.thr_hover = math::constrain(_params.thr_hover, _params.thr_min, _params.thr_max);
-		param_get(_params_handles.tilt_max_air, &_params.tilt_max_air);
-		_params.tilt_max_air = math::radians(_params.tilt_max_air);
-		param_get(_params_handles.land_speed, &_params.land_speed);
-		param_get(_params_handles.tko_speed, &_params.tko_speed);
-		param_get(_params_handles.tilt_max_land, &_params.tilt_max_land);
-		_params.tilt_max_land = math::radians(_params.tilt_max_land);
-
-		float v;
-		int32_t v_i;
-		param_get(_params_handles.xy_p, &v);
-		_params.pos_p(0) = v;
-		_params.pos_p(1) = v;
-		param_get(_params_handles.z_p, &v);
-		_params.pos_p(2) = v;
-		param_get(_params_handles.xy_vel_p, &v);
-		_params.vel_p(0) = v;
-		_params.vel_p(1) = v;
-		param_get(_params_handles.z_vel_p, &v);
-		_params.vel_p(2) = v;
-		param_get(_params_handles.xy_vel_i, &v);
-		_params.vel_i(0) = v;
-		_params.vel_i(1) = v;
-		param_get(_params_handles.z_vel_i, &v);
-		_params.vel_i(2) = v;
-		param_get(_params_handles.xy_vel_d, &v);
-		_params.vel_d(0) = v;
-		_params.vel_d(1) = v;
-		param_get(_params_handles.z_vel_d, &v);
-		_params.vel_d(2) = v;
-		param_get(_params_handles.z_vel_max_up, &v);
-		_params.vel_max_up = v;
-		param_get(_params_handles.z_vel_max_down, &v);
-		_params.vel_max_down = v;
-		param_get(_params_handles.xy_vel_max, &v);
-		_params.vel_max_xy = v;
-		param_get(_params_handles.xy_vel_cruise, &v);
-		_params.vel_cruise_xy = v;
-		param_get(_params_handles.hold_max_xy, &v);
-		_params.hold_max_xy = math::max(0.f, v);
-		param_get(_params_handles.hold_max_z, &v);
-		_params.hold_max_z = math::max(0.f, v);
-		param_get(_params_handles.rc_flt_smp_rate, &(_params.rc_flt_smp_rate));
-		_params.rc_flt_smp_rate = math::max(1.0f, _params.rc_flt_smp_rate);
-		/* since we use filter to detect manual direction change, take half the cutoff of the stick filtering */
-		param_get(_params_handles.rc_flt_cutoff, &(_params.rc_flt_cutoff));
+		ModuleParams::updateParams();
+		SuperBlock::updateParams();
+
+		_flight_tasks.handleParameterUpdate();
+
+		/* initialize vectors from params and enforce constraints */
+
+		_pos_p(0) = _xy_p.get();
+		_pos_p(1) = _xy_p.get();
+		_pos_p(2) = _z_p.get();
+
+		_vel_p(0) = _xy_vel_p.get();
+		_vel_p(1) = _xy_vel_p.get();
+		_vel_p(2) = _z_vel_p.get();
+
+		_vel_i(0) = _z_vel_i.get();
+		_vel_i(1) = _z_vel_i.get();
+		_vel_i(2) = _z_vel_i.get();
+
+		_vel_d(0) = _xy_vel_d.get();
+		_vel_d(1) = _xy_vel_d.get();
+		_vel_d(2) = _z_vel_d.get();
+
+		_thr_hover.set(math::constrain(_thr_hover.get(), _thr_min.get(), _thr_max.get()));
```

Comment:
`.set()` only sets the parameter locally temporarily right? Altought if it's out of bounds it could be saved back as feedback for the user that the velue he changed actually doesn't apply.

Reply:
Correct. I thought about it, and we could write it back, but I did not want to change the behavior in this PR.

Reply:
What does that mean locally? It I there are two modules that both consume the same parameter, and in one of the parameter I use the setter method, does the other module then have the new value or the default? 
Or another scenario. If there is a class member that also uses the same parameter as the base class and the base class overwrites one parameter with the setter method, what is the value of the parameter that the member class will consume?

Reply:
> If I there are two modules that both consume the same parameter, and in one of the modules I use the setter method for that parameter, does the other module then have the new value or the default?

It keeps the previous value. You can use the `commit()` method to store it persistently and notify all modules about the param change.

> If there is a class member that also uses the same parameter as the base class and the base class overwrites one parameter with the setter method, what is the value of the parameter that the member class will consume?

If it accesses the same class member, it will have the updated value. If you define the same param in the subclass as a different class member, it's separate with the same behavior as in the separate modules case.

Reply:
> If you define the same param in the subclass as a different class member, it's separate with the same behavior as in the separate modules case.

good to know. thanks